official/resnet/imagenet_test.py

```diff
@@ -46,11 +47,12 @@ def reshape(shape):
 
     with graph.as_default(), self.test_session(
         use_gpu=with_gpu, force_gpu=with_gpu):
-      model = resnet_model.imagenet_resnet_v2(
-          resnet_size, 456,
+      model = imagenet_main.ImagenetModel(
+          resnet_size,
           data_format='channels_first' if with_gpu else 'channels_last')
+      model.num_classes = 456
```

Comment:
Changing `num_classes` after the fact seems a bit iffy to me as it isn't obvious that it is safe to do so (for example, does the base-class of `ImagenetModel` set some state based on `num_classes` provided to it?).

I would suggest that instead of changing this after the fact, we make `ImagenetModel.__init__` take a `num_classes` argument (that can default to `_NUM_CLASSES`). So line 50 will become:

```python
model = imagenet_main.ImagenetModel(
  resnet_size,
  data_format=...,
  num_classes=456)
```

Reply:
This appeared just to be a relic preserved from the previous test, and not really testing anything in particular. Just used the default class size, rather than introducing a new param for the sake of one test.